system/jlib/jstats.cpp

```diff
@@ -1915,15 +1995,33 @@ class CStatisticCollection : public CInterfaceOf<IStatisticCollection>
         for (iter.first(); iter.isValid(); iter.next())
             iter.query().getMinMaxActivity(minValue, maxValue);
     }
+    virtual bool setStatistic(const char *scope, StatisticKind kind, unsigned __int64 value) override
```

Comment:
this is essentially a 'ensureScope' function + a scope->setStatisitic(kind, value); implementation.
It would be cleaner (and enable other changes) if it was separated out to:
```
IStatisticCollecton::ensureScope(const char *fullScope) = 0;
void setStatistic(StatisticKind kind, unsigned __int64 value) = 0;
```

Although at that point, setStatistic, is very much like updateStatistic, so it may be better to just expose updateStatistic and have the aggregator class call it directly.